lightning/restore/restore.go

```diff
@@ -304,48 +305,207 @@ outside:
 	return errors.Trace(err)
 }
 
-func (rc *RestoreController) restoreSchema(ctx context.Context) error {
-	if !rc.cfg.Mydumper.NoSchema {
-		if rc.tidbGlue.OwnsSQLExecutor() {
-			db, err := DBFromConfig(rc.cfg.TiDB)
-			if err != nil {
-				return errors.Annotate(err, "connect to tidb failed")
-			}
-			defer db.Close()
-			db.ExecContext(ctx, "SET SQL_MODE = ?", rc.cfg.TiDB.StrSQLMode)
-		}
+type schemaStmtType int
 
-		for _, dbMeta := range rc.dbMetas {
-			task := log.With(zap.String("db", dbMeta.Name)).Begin(zap.InfoLevel, "restore table schema")
+const (
+	schemaCreateDatabase = iota
+	schemaCreateTable
+	schemaCreateView
+)
+
+type schemaJob struct {
+	dbName string
+	stmts  []*schemaStmt
```

Comment:
Since the `tblName`, `stmeType` is always the same in `stmts`, I think we can change this struct into : 
```
type schemaJob struct {
  dbName string
  tblNmae string // empty for create db jobs
  stmtType schemaStmtType
  stmts []string
}
```

Reply:
make sense